### PR TITLE
MQY-738: replaced diesel with sqlx

### PR DESCRIPTION
**Pull request by nicholasatmerqury**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ license = "AGPL-3.0-only"
 actix-tls = "3.1"
 actix-web = { version = "4", features = ["openssl"] }
 base64 = "0.21.4"
-diesel = { version = "2.1.2", features = ["postgres", "uuid"] }
+sqlx = { version = "0.7.3", features = ["postgres", "runtime-tokio", "tls-rustls", "uuid", "chrono"] }
```
Comment (nicholasatmerqury) [inline Cargo.toml:14]: *Observation:* Consider listing dependencies in alphabetical order.

File: migrations/20230113082423_create_keys.sql
Comment (nicholasatmerqury) [inline migrations/20230113082423_create_keys.sql:1]: *Observations:* This was not introduced in this changeset, however, kindly consider the following:

- Fix typo `last_modofied_at`.
- Add an index on the `id` column.

File: src/ops/key.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use crate::models::key::NewKey;
 use crate::{converter, db};
 use crate::{error::Error, models::key::Key};
 use actix_web::http::StatusCode;
-use diesel::prelude::*;
+use sqlx::PgPool;
```
Comment (nicholasatmerqury) [inline src/ops/key.rs:8]: *Suggestion:* Consider alphabetical order.

File: src/ops/key.rs [hunk 3]
```diff
@@ -115,57 +113,69 @@ pub fn save_keys(
...
-        }
-        Err(e) => {
-            error!("Failed to save records to db: {:?}", e);
-            Err(Error::internal_server_error())
-        }
+    let pool = &db::establish_connection().await?;
+    let mut num_inserted_rows: u64 = 0;
+    for key in keys_to_insert{
+        let result = match sqlx::query!(
+            r#"INSERT INTO keys (id, master_sae_id, slave_sae_id, size, content)VALUES ($1, $2, $3, $4, $5);"#,
```
Comment (nicholasatmerqury) [inline src/ops/key.rs:120]: *Suggestion:* Here and in similar places, if it adds any value, consider using the sqlx macro `sqlx::query_file_as!`. This allows for segregating the SQL statements into dedicated files, taking them out of Rust code.

File: src/ops/key.rs [hunk 4]
```diff
@@ -115,57 +113,69 @@ pub fn save_keys(
...
-            Ok(())
-        }
-        Err(e) => {
-            error!("Failed to save records to db: {:?}", e);
-            Err(Error::internal_server_error())
-        }
+    let pool = &db::establish_connection().await?;
+    let mut num_inserted_rows: u64 = 0;
+    for key in keys_to_insert{
+        let result = match sqlx::query!(
```
Comment (nicholasatmerqury) [inline src/ops/key.rs:119]: *Nitpick:* It seems your IDE is not properly configured RE Rust code formatting. Noticed this due to the missing space after `keys_to_insert` and in similar places below. Can help out with configuring your IDE, if needed.

File: src/ops/key.rs [hunk 5]
```diff
@@ -115,57 +113,69 @@ pub fn save_keys(
...
+            id = $1 AND
+            master_sae_id = $2 AND
+            active = TRUE
+        ;"#,
+        key_id,
+        master_sae_id,
+    )
+    .fetch_one(pool)
+    .await{
         Ok(res) => res,
```
Comment (nicholasatmerqury) [inline src/ops/key.rs:179]: *Suggestion:* If it aids code readability, consider retrieving the `count` field here.

File: src/ops/key.rs (hunk 3 above)
Comment (nicholasatmerqury) [reply src/ops/key.rs]: Good idea. Thanks for following-up, resolving this thread.

**Review by nicholasatmerqury — COMMENTED** [review]
Thanks for following-up. 

One final thing please, consider adding the `SQLX_OFFLINE=true` flag to the `run_tests` Makefile target.

Excellent work 🙌

**Review by danielatmerqury — APPROVED** [review]

**Pull request merged**